docs/root/configuration/http/http_conn_man/headers.rst

```diff
@@ -341,6 +357,14 @@ It is a common case where a service wants to know what the originating protocol
 of the connection terminated by front/edge Envoy. *x-forwarded-proto* contains this information. It
 will be set to either *http* or *https*.
 
+Downstream *x-forwarded-proto* headers will only be trusted if *xff_num_trusted_hops* is non-zero.
+If *xff_num_trusted_hops* is zero, downstream *x-forwarded-proto* headers will be set to http
+or https based on if the downstream connection is TLS or not.
```

Comment:
Per above, for H1 scheme will also be set in this case, right?

Reply:
yes, though I was trying to talk to scheme updates in the scheme section and XFP in the XFP section.  Given the complexity do you think redundancy is a better plan?

Reply:
Yeah I would be pretty redundant here and over share, given how important this is to understand.